File: core/usecase/Singin.py
from data.model.UserModel import UserModel

def usecase_singing(dni, password, email, name, lastname, employee_number, db):
    _validator(dni, password, email, name, lastname)
    user = UserModel(
            dni=int(dni),
            password= password,
            email= email,
            name= name,
            lastname= lastname,
            employee_number= employee_number
        )
    db.insert_user(dni, user)
    return True

def _validator(dni, password, email, name, lastname):
    if dni == "":
        raise Exception()
    if password == "":
        raise Exception()
    if email == "":
        raise Exception()
    if name == "":
        raise Exception()
    if lastname == "":
        raise Exception()

File: templates/scripts/singin.js
function singin(){
    var dni = document.getElementById("singin-input-dni")
    var password = document.getElementById("singin-inpunt-password")
    var email = document.getElementById("singin-inpunt-email")
    var name = document.getElementById("singin-inpunt-name")
    var lastname = document.getElementById("singin-inpunt-lastname")
    var employee_number = 0

    fetch(
        "/singin",
        {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify(
                {
                    dni: dni,
                    password: password,
                    email: email,
                    name: name,
                    lastname: lastname,
                    employee_number: employee_number
                }
            )
        }
    ).then(response => response.json())
}